carouselrecyclerview/src/main/java/com/jackandphantom/carouselrecyclerview/view/ReflectionViewContainer.kt

```diff
@@ -0,0 +1,270 @@
+import android.graphics.Paint
+import android.graphics.PorterDuffXfermode
+import android.graphics.Shader
+import android.graphics.PorterDuff
+import android.graphics.Canvas
+import android.os.Build
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import com.jackandphantom.carouselrecyclerview.R
+
+
+/**
+ * This is an extension of LinearLayout to have reflection of child view in it.
+ * To keep the layout simple i have restricted the layout to contain only one child.
+ * The purpose of this layout is to have a live reflection of the view in it.
+ * If more than one view is present in the layout, it only generates the reflection of the first view.
+ * All the draw event are also propagated to the reflection view, so it take twice the time to render. :(
+ * */
+
+class ReflectionViewContainer : LinearLayout {
+
+    companion object {
+        // Relative height of the view to generate reflection aka Reflection Depth/Length
+        private const val DEFAULT_RELATIVE_DEPTH = 0.5f
+
+        // Gap b/w the view and the reflection formed
+        // Same effect can be achieved with padding but the distance will be twice
+        private const val DEFAULT_GAP = 0.0f
+    }
+
+    private lateinit var mReflect: Reflect
+    private var mRelativeDepth: Float = DEFAULT_RELATIVE_DEPTH
+    private var mReflectionGap: Float = DEFAULT_GAP
+
+    // constructor for programmatically adding view
+    constructor(context: Context, view: View) : super(context, null, R.attr.reflect_reflectionLayoutStyle) {
+        addView(view)
+        mReflect = Reflect(this.context).apply {
+            setupView(view, mReflectionGap, mRelativeDepth)
+            alpha = 0.85f
+        }
+        addView(mReflect)
+        initLayout(null, R.attr.reflect_reflectionLayoutStyle)
+    }
+
+    constructor(context: Context) : super(context, null, R.attr.reflect_reflectionLayoutStyle) {
+        initLayout(null, R.attr.reflect_reflectionLayoutStyle)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(
+        context,
+        attrs,
+        R.attr.reflect_reflectionLayoutStyle
+    ) {
+        initLayout(attrs, R.attr.reflect_reflectionLayoutStyle)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        initLayout(attrs, defStyleAttr)
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int,
+        defStyleRes: Int
+    ) : super(
+        context,
+        attrs,
+        defStyleAttr,
+        defStyleRes
+    ) {
+        initLayout(attrs, defStyleAttr)
+    }
+
+    private fun initLayout(attrs: AttributeSet?, defStyleAttr: Int) {
+        val a = context.obtainStyledAttributes(
+            attrs,
+            R.styleable.ReflectionViewContainer,
```

Comment:
Hi, @snehilrx  I don't see this styleable in attrs.xml, so which style you are using here?